marketplaceItems/audioFocus/appResources/appData/audioFocus_app.js

```diff
@@ -0,0 +1,459 @@
+
+                // add to the real position so the grab-point is out in front of the hand, a bit
+                if (doOffset) {
+                    var offset = getGrabPointSphereOffset(handController);
+                    position = Vec3.sum(position, Vec3.multiplyQbyV(orientation, offset));
+                }
+
+            } else if (!HMD.isHandControllerAvailable()) {
+                // NOTE: keep this offset in sync with scripts/system/controllers/handControllerPointer.js:493
+                var VERTICAL_HEAD_LASER_OFFSET = 0.1 * MyAvatar.sensorToWorldScale;
+                position = Vec3.sum(Camera.position, Vec3.multiplyQbyV(Camera.orientation, { x: 0, y: VERTICAL_HEAD_LASER_OFFSET, z: 0 }));
+                orientation = Quat.multiply(Camera.orientation, Quat.angleAxis(-90, { x: 1, y: 0, z: 0 }));
+                valid = true;
+            }
+        }
+
+        return {
+            position: position,
+            translation: position,
+            orientation: orientation,
+            rotation: orientation,
+            valid: valid
+        };
+    }
+
+    // #endregion
+    // *************************************
+    // END UTILITY FUNCTIONS
+    // *************************************
+
+
+    // *************************************
+    // START INIT
+    // *************************************
+    // #region Init
+
+    var isEnabled = true;
+    var soloAvatars = {};
+
+    // #endregion
+    // *************************************
+    // STOP INIT
+    // *************************************
+
+
+    // *************************************
+    // START MAPPING FUNCTIONS
+    // *************************************
+    // #region Mapping
+
+    var lastMouseX;
+    var lastMouseY;
+    // Record the last mousePressEvent
+    function mousePressEvent(event) {
+        lastMouseX = event.x;
+        lastMouseY = event.y;
+    }
+
+
+    // Enables all the pointers created and the mapping
+    function enable() {
+        Controller.enableMapping(MAPPING_NAME);
+    }
+
+
+    // Disables all the pointers created and the mapping 
+    function disable() {
+        Controller.disableMapping(MAPPING_NAME);
+    }
+
+
+    // Returns the right UUID based on hand triggered
+    function getUUIDFromLaser(hand){
+        hand = hand === Controller.Standard.LeftHand 
+            ? Controller.Standard.LeftHand
+            : Controller.Standard.RightHand;
+        
+        var pose = getControllerWorldLocation(hand);
+        var start = pose.position;
+        var direction = Vec3.multiplyQbyV(pose.orientation, {x:0, y:1, z: 0});
+
+        var avatarIntersection = AvatarList.findRayIntersection({origin:start, direction:direction});
+        
+        var uuid = avatarIntersection.avatarID;
+
+        return uuid;
+    }
+
+
+    var MAPPING_NAME = "SOLO_POINTER";
+    var mapping = Controller.newMapping(MAPPING_NAME);
+
+    mapping.from(Controller.Hardware.Keyboard.LeftMouseButton).to(function (value) {
```

Comment:
Why do you still need this mapping if you're using `mousePressEvent`? Is there a reason you couldn't put the contents of this function (minus what's on lines 171-173) inside `mousePressEvent` above?

Reply:
if you did that, you'd have to disconnect and reconnect `mousePressEvent` when you enable/disable the controller mappings too